src/main/java/net/minestom/server/item/ItemTag.java

```diff
@@ -1,112 +1,65 @@
 package net.minestom.server.item;
 
+import net.minestom.server.tag.Tag;
 import org.jetbrains.annotations.NotNull;
 import org.jglrxavpok.hephaistos.nbt.NBT;
 import org.jglrxavpok.hephaistos.nbt.NBTCompound;
-import org.jglrxavpok.hephaistos.nbt.NBTList;
 
 import java.util.function.BiConsumer;
 import java.util.function.Function;
 
-public class ItemTag<T> {
+/**
+ * @deprecated use {@link Tag}.
+ */
+@Deprecated
+public class ItemTag<T> extends Tag<T> {
 
-    private final String key;
-    private final Function<NBTCompound, T> readFunction;
-    private final BiConsumer<NBTCompound, T> writeConsumer;
-
-    private ItemTag(@NotNull String key,
-                    @NotNull Function<NBTCompound, T> readFunction,
-                    @NotNull BiConsumer<NBTCompound, T> writeConsumer) {
-        this.key = key;
-        this.readFunction = readFunction;
-        this.writeConsumer = writeConsumer;
-    }
-
-    public @NotNull String getKey() {
-        return key;
-    }
-
-    protected T read(@NotNull NBTCompound nbtCompound) {
-        return readFunction.apply(nbtCompound);
+    protected ItemTag(@NotNull String key, @NotNull Function<NBTCompound, T> readFunction, @NotNull BiConsumer<NBTCompound, T> writeConsumer) {
+        super(key, readFunction, writeConsumer);
     }
 
-    protected void write(@NotNull NBTCompound nbtCompound, @NotNull T value) {
-        this.writeConsumer.accept(nbtCompound, value);
+    public static @NotNull Tag<Byte> Byte(@NotNull String key) {
+        return Tag.Byte(key);
     }
 
-    public static @NotNull ItemTag<Byte> Byte(@NotNull String key) {
-        return new ItemTag<>(key,
-                nbtCompound -> nbtCompound.getByte(key),
-                (nbtCompound, value) -> nbtCompound.setByte(key, value));
+    public static @NotNull Tag<Short> Short(@NotNull String key) {
```

Comment:
This will be a breaking change - is there any chance we could have `ItemTag` be some sort of fake wrapper around `Tag`? Or get this to return `ItemTag`?